hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/GrpcBlockItemWriter.java

```diff
@@ -0,0 +1,207 @@
+            @Override
+            public void onNext(PublishStreamResponse streamResponse) {
+                if (streamResponse.hasAcknowledgement()) {
+                    final Acknowledgement acknowledgement = streamResponse.getAcknowledgement();
+                    if (acknowledgement.hasBlockAck()) {
+                        logger.info("PublishStreamResponse: a full block received: {}", acknowledgement.getBlockAck());
+                    } else if (acknowledgement.hasItemAck()) {
+                        logger.info(
+                                "PublishStreamResponse: a single block item is received: {}",
+                                acknowledgement.getItemAck());
+                    }
+                }
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                // Maybe this should be considered in this case:
+                // https://github.com/hashgraph/hedera-services/issues/15530
+                final Status status = fromThrowable(t);
+                logger.error(status.toString());
+            }
+
+            @Override
+            public void onCompleted() {
+                try {
+                    if (channel != null) {
+                        channel.shutdown().awaitTermination(10, TimeUnit.MILLISECONDS);
+                        channel = null;
+                    }
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                logger.info("PublishStreamResponse completed");
+            }
+        });
+
+        this.state = State.OPEN;
+    }
+
+    @Override
+    public BlockItemWriter writeItem(@NonNull Bytes serializedItem) {
+        if (state != State.OPEN) {
+            throw new IllegalStateException(
+                    "Cannot write to a GrpcBlockItemWriter that is not open for block: " + this.blockNumber);
+        }
+
+        PublishStreamRequest request = PublishStreamRequest.newBuilder().build();
+        try {
+            request = PublishStreamRequest.newBuilder()
+                    .setBlockItem(BlockItem.parseFrom(serializedItem.toByteArray()))
+                    .build();
+            requestObserver.onNext(request);
+        } catch (InvalidProtocolBufferException e) {
+            final String message = INVALID_MESSAGE.formatted("PublishStreamResponse", request);
+            throw new RuntimeException(message, e);
+        }
+        return this;
+    }
+
+    @Override
+    public void closeBlock() {
+        if (state.ordinal() < GrpcBlockItemWriter.State.OPEN.ordinal()) {
+            throw new IllegalStateException("Cannot close a GrpcBlockItemWriter that is not open");
+        } else if (state.ordinal() == GrpcBlockItemWriter.State.CLOSED.ordinal()) {
+            throw new IllegalStateException("Cannot close a GrpcBlockItemWriter that is already closed");
+        }
+
+        requestObserver.onCompleted();
```

Comment:
I like the request observer, it's cleaner than checking for the `FILE` type and closing inside this class. Well done!